aws-crt-http/HttpConnection.cs

```diff
@@ -148,16 +135,25 @@ public sealed class HttpRequest
         public string Method { get; set; }
         public string Uri { get; set; }
         public HttpHeader[] Headers { get; set; }
-        public event EventHandler<StreamOutgoingBodyEventArgs> StreamOutgoingBody;
+
+        public Stream BodyStream { get; set; }

```

Comment:
Should we throw an exception if you try to set this and it can't seek? or do we just let it throw when you try to seek in an unsupported way later? If so, can we catch it and re-frame the exception so we explain what happened to the user, because it's going to be super confusing at first.

Reply:
I don't think we should throw here, but we should throw if you try and sign a request that can't seek.